### ISSUE TITLE
[salt-cloud] 2015.8.3 Provisioning VMs in Parallel on CentOS 6 fails

### ISSUE THREAD

**Issue by mbochenk** [saltstack/salt#29481]
Provisioning via salt-cloud with "-P" fails on CetOS 6.7:
```
16:09:04			  File "/usr/lib64/python2.6/multiprocessing/process.py", line 232, in _bootstrap
16:09:04			    self.run()
16:09:04			  File "/usr/lib64/python2.6/multiprocessing/process.py", line 88, in run
16:09:04			    self._target(*self._args, **self._kwargs)
16:09:04			  File "/usr/lib/python2.6/site-packages/salt/cloud/__init__.py", line 1189, in create
16:09:04			    self.opts
16:09:04			  File "/usr/lib/python2.6/site-packages/salt/utils/cloud.py", line 159, in gen_keys
16:09:04			    salt.crypt.gen_keys(tdir, 'minion', keysize)
16:09:04			  File "/usr/lib/python2.6/site-packages/salt/crypt.py", line 88, in gen_keys
16:09:04			    gen = RSA.generate(bits=keysize, e=65537)
16:09:04			  File "/usr/lib64/python2.6/site-packages/Crypto/PublicKey/RSA.py", line 508, in generate
16:09:04			    obj = _RSA.generate_py(bits, rf, progress_func, e)    # TODO: Don't use legacy _RSA module
16:09:04			  File "/usr/lib64/python2.6/site-packages/Crypto/PublicKey/_RSA.py", line 50, in generate_py
16:09:04			    p = pubkey.getStrongPrime(bits>>1, obj.e, 1e-12, randfunc)
16:09:04			  File "/usr/lib64/python2.6/site-packages/Crypto/Util/number.py", line 265, in getStrongPrime
16:09:04			    randfunc)
16:09:04			  File "/usr/lib64/python2.6/site-packages/Crypto/Random/_UserFriendlyRNG.py", line 202, in read
16:09:04			    return self._singleton.read(bytes)
16:09:04			  File "/usr/lib64/python2.6/site-packages/Crypto/Random/_UserFriendlyRNG.py", line 178, in read
16:09:04			    return _UserFriendlyRNG.read(self, bytes)
16:09:04			  File "/usr/lib64/python2.6/site-packages/Crypto/Random/_UserFriendlyRNG.py", line 132, in read
16:09:04			    retval = self._fa.random_data(N)
16:09:04			  File "/usr/lib64/python2.6/site-packages/Crypto/Random/Fortuna/FortunaAccumulator.py", line 148, in random_data
16:09:04			    return self.generator.pseudo_random_data(bytes)
16:09:04			  File "/usr/lib64/python2.6/site-packages/Crypto/Random/Fortuna/FortunaGenerator.py", line 93, in pseudo_random_data
16:09:04			    for i in xrange(num_full_blocks):
16:09:04			  File "/usr/lib64/python2.6/site-packages/Crypto/Random/_UserFriendlyRNG.py", line 202, in read
16:09:04			    return self._singleton.read(bytes)
16:09:04			  File "/usr/lib64/python2.6/site-packages/Crypto/Random/_UserFriendlyRNG.py", line 178, in read
16:09:04			    return _UserFriendlyRNG.read(self, bytes)
16:09:04			  File "/usr/lib64/python2.6/site-packages/Crypto/Random/_UserFriendlyRNG.py", line 132, in read
16:09:04			    retval = self._fa.random_data(N)
16:09:04			  File "/usr/lib64/python2.6/site-packages/Crypto/Random/Fortuna/FortunaAccumulator.py", line 148, in random_data
16:09:04			    return self.generator.pseudo_random_data(bytes)
16:09:04			  File "/usr/lib64/python2.6/site-packages/Crypto/Random/Fortuna/FortunaGenerator.py", line 93, in pseudo_random_data
16:09:04			    for i in xrange(num_full_blocks):
16:09:04			  File "/usr/lib64/python2.6/site-packages/Crypto/Random/_UserFriendlyRNG.py", line 202, in read
16:09:04			    return self._singleton.read(bytes)
16:09:04			  File "/usr/lib64/python2.6/site-packages/Crypto/Random/_UserFriendlyRNG.py", line 178, in read
16:09:04			    return _UserFriendlyRNG.read(self, bytes)
16:09:04			  File "/usr/lib64/python2.6/site-packages/Crypto/Random/_UserFriendlyRNG.py", line 137, in read
16:09:04			    self._check_pid()
16:09:04			  File "/usr/lib64/python2.6/site-packages/Crypto/Random/_UserFriendlyRNG.py", line 153, in _check_pid
16:09:04			    raise AssertionError("PID check failed. RNG must be re-initialized after fork(). Hint: Try Random.atfork()")
16:09:04			AssertionError: PID check failed. RNG must be re-initialized after fork(). Hint: Try Random.atfork()
```

This seems to affect only salt-cloud and related to master only - minions with 2015.8.3 work fine and are generated via 2015.5.8

Master:
```
           Salt: 2015.5.8
         Python: 2.6.6 (r266:84292, Jul 23 2015, 15:22:56)
         Jinja2: 2.2.1
       M2Crypto: 0.20.2
 msgpack-python: 0.4.6
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.11
          ioflo: Not Installed
          PyZMQ: 14.5.0
           RAET: Not Installed
            ZMQ: 4.0.5
           Mako: Not Installed
        Tornado: 4.2.1
        timelib: Not Installed
       dateutil: Not Installed
```
Minion:
```
Salt Version:
           Salt: 2015.8.3

Dependency Versions:
         Jinja2: 2.8
       M2Crypto: 0.20.2
           Mako: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
         Python: 2.6.6 (r266:84292, Jul 23 2015, 15:22:56)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: 0.2.4

System Versions:
           dist: centos 6.7 Final
        machine: x86_64
        release: 2.6.32-573.8.1.el6.x86_64
         system: CentOS 6.7 Final
```

This happens to yum installation on master, stable via bootstrap or git via bootstrap. 
This seems very similar to closed issue https://github.com/saltstack/salt/issues/24176

**Comment by mbochenk** [comment]
Might be not related, but in non-parallel mode salt-cloud cannot authenticate via RSA key:
```
17:36:26			Permission denied (publickey,gssapi-keyex,gssapi-with-mic,password).
```

**Comment by m-rouget** [comment]
Got the same issue on Debian 8.2 (I use the _debian-8_ image from GCE)

Master:
```
Salt Version:
           Salt: 2015.8.3

Dependency Versions:
         Jinja2: 2.7.3
       M2Crypto: Not Installed
           Mako: 1.0.0
         PyYAML: 3.11
          PyZMQ: 14.4.0
         Python: 2.7.9 (default, Mar  1 2015, 12:57:24)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.2
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: 0.8.2
        timelib: Not Installed

System Versions:
           dist: debian 8.2
        machine: x86_64
        release: 3.16.0-4-amd64
         system: debian 8.2
```

**Comment by m-rouget** [comment]
salt-cloud with -P *and* a mapfile works without issues:
```sudo salt-cloud -m salt-my-config.mapfile -l debug -y -P```

Instances are created in parallel and deployed with success.

**Comment by jfindlay** [comment]
@mbochenk, @m-rouget, thanks for the report.

**Comment by rallytime** [comment]
I suspect this is a duplicate of #28396. I'm going to close this in favor of the older issue. Please feel free to add any additional comments about this issue there.
